src/pharmpy/tools/amd/run.py

```diff
@@ -999,6 +974,47 @@ def _run_mechanistic_exploratory_covariates(model, modelfit_results):
     return _run_mechanistic_exploratory_covariates
 
 
+def _mechanistic_cov_extraction(search_space, model, mechanistic_covariates):
+    mechanistic_covariates = [c if isinstance(c, str) else set(c) for c in mechanistic_covariates]
+    # Extract them and all forced
+    mfl = ModelFeatures.create_from_mfl_statement_list(search_space)
+    mfl_covariates = mfl.expand(model).covariate
+    mechanistic_searchspace = []
+    for cov_statement in mfl_covariates:
+        if not cov_statement.optional.option:
+            # Not optional -> Add search space (was added in structural run)
+            mechanistic_searchspace.append(cov_statement)
+        else:
+            current_cov = []
+            current_param = []
+            for cov in cov_statement.covariate:
+                if cov in mechanistic_covariates:
+                    current_cov.append(cov)
+                    current_param.append(cov_statement.parameter)
+                else:
+                    for param in cov_statement.parameter:
+                        if {cov, param} in mechanistic_covariates:
```

Comment:
Maybe misunderstanding but if the set-conversion happens at line 978 why is it needed here too? Could line 978 be removed and just this kept?

Reply:
Ah okay, I understand now, makes sense!